### PR TITLE
Throw an error when no identifier is defined

### PR DESCRIPTION
**Pull request by soyuka**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | kinda
| Deprecations? | no
| Tickets       | 
| License       | MIT
| Doc PR        | api-platform/docs#...

### CODE REVIEW

File: src/Api/IdentifiersExtractor.php [hunk 1]
```diff
@@ -57,6 +57,10 @@ public function getIdentifiersFromResourceClass(string $resourceClass): array
             }
         }
 
+        if (!$identifiers) {
+            throw new RuntimeException(sprintf('No identifier found in "%s"', $resourceClass));
```
Comment (dunglas) [inline src/Api/IdentifiersExtractor.php:61]:
```suggestion
            throw new RuntimeException(sprintf('No identifier defined "%s". You should add #[\ApiPlatform\Core\Annotation\ApiProperty(identifier: true)]" on the property identifying the resource.', $resourceClass));
```

**Review by dunglas — CHANGES_REQUESTED** [review]

File: src/Api/IdentifiersExtractor.php [hunk 2]
```diff
@@ -57,6 +57,10 @@ public function getIdentifiersFromResourceClass(string $resourceClass): array
             }
         }
 
+        if (!$identifiers) {
+            throw new RuntimeException(sprintf('No identifier defined "%s". You should add #[\ApiPlatform\Core\Annotation\ApiProperty(identifier: true)]" on the property identifying the resource."', $resourceClass));
```
Comment (maks-rafalko) [inline src/Api/IdentifiersExtractor.php:61]: What do you think about 

```diff
- No identifier defined "%s".
+ No identifier defined in "%s".
```

I mean, "in" is missed from my point of view

**Comment by julienfalque** [comment]
This new exception seems to break POST/PUT endpoints that use a DTO without id as the resource. Is that intended? If so, what are the alternatives?

**Comment by soyuka** [comment]
It should not break unless your DTO is a resource. Please open an issue and provide a reproducer I definitely want to check this!

**Comment by maks-rafalko** [comment]
@soyuka there is already an issue https://github.com/api-platform/core/issues/3975

Please re-read it, author has edited description

**Pull request merged**